### PR TITLE
Ignore Deployment Restriction for WebSocket 2.1,  Added websocket deployment tests

### LINKED ISSUE #22364: Add Wsoc 2.1 Test to verify endpoints can be added outside of deployment phase.
WebSocket 2.1 Spec: `Remove the restriction that, in a Jakarta web container environment, endpoints can only registered during the deployment of the web application.` 

See #20584

Turns out that restriction wasn't removed.

### PR DESCRIPTION
**Pull request by isaacrivriv**
- Made changes to include functionality to ignore restriction of adding endpoints after deployment
- Added test to verify deployment restrictions

fixes #22364

### CODE REVIEW

**Comment by isaacrivriv** [comment]
#build

**Review by volosied — DISMISSED** [review]
Looks good to me. Thanks for getting this up!

File: dev/io.openliberty.wsoc.internal_fat/fat/src/io/openliberty/wsoc/tests/BasicTest.java [hunk 1]
```diff
@@ -216,6 +224,30 @@ public void testURIOverlappingWithHtml() throws Exception {
...
+        // New test for WebSocket 2.1 change to allow adding endpoints outside of initialization/deployment phase of web application https://github.com/jakartaee/websocket/issues/211
+        
+        // Call to open the connection to the servlet. Need to add getResponseCode to actually complete the connection but test doesn't care about the response
+        // We just check the logs for the expected new endpoint added or not
+        HttpUtils.getHttpConnectionWithAnyResponseCode(LS, "/basic/addEndpoint").getResponseCode();
+        
+        // This test causes SRVE0777E: Exception thrown by application class ServerContainerExt.addEndpoint which is expected
+        // Also cause SRVE0315E An exception occurred: java.lang.Throwable: java.lang.IllegalStateException: endpoint.addsclosed also expected
+        // and SRVE8094W WARNING: Cannot set header. Response already committed to show up
+        if(RepeatTestFilter.isRepeatActionActive(JakartaEE10Action.ID)){
```
Comment (pnicolucci) [inline dev/io.openliberty.wsoc.internal_fat/fat/src/io/openliberty/wsoc/tests/BasicTest.java]: Any reason you didn't just do `JakartaEE10Action.isActive()` ?

File: dev/io.openliberty.wsoc.internal_fat/test-applications/basic.war/src/basic/war/servlet/AddEndpointServlet.java [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+import basic.war.EchoServerEP;
+
+/**
+ * Servlet for testing removed restriction when adding an endpoint in Websocket 2.1 with addEndpoint methods in ServerContainer https://github.com/jakartaee/websocket/issues/211
+ * As stated in the specification before 2.1, "These methods are only operational during the application deployment phase of an application. Specifically, as soon as any of the
+ * server endpoints within the application have accepted an opening handshake request, the APIs may not longer be used. This restriction may 
+ * be relaxed in a future version."
+ * 
+ * Before 2.1, endpoints can only registered during the deployment of the web application. Once a websocket request enters the webcontainer,
+ * no more endpoints could be added with the API as stated in the spec "are only operational during the application deployment phase of an application. Specifically, as soon as any of the server endpoints within the application have accepted an opening handshake request, the apis may not longer be used. This restriction may be relaxed in a future version.". This restriction was removed in EE10 WebSocket 2.1
```
Comment (pnicolucci) [inline dev/io.openliberty.wsoc.internal_fat/test-applications/basic.war/src/basic/war/servlet/AddEndpointServlet.java]: Can you do this line over multiple lines? It seems super long relative to the rest of the javadoc

File: dev/io.openliberty.wsoc.internal_fat/fat/src/io/openliberty/wsoc/tests/BasicTest.java (hunk 1 above)
Comment (isaacrivriv) [reply dev/io.openliberty.wsoc.internal_fat/fat/src/io/openliberty/wsoc/tests/BasicTest.java]: No specific reason, I was working with the RepeatTestFilter lately and just went with it here. I can change it back to make it consistent across all the tests

File: dev/io.openliberty.wsoc.internal_fat/test-applications/basic.war/src/basic/war/servlet/AddEndpointServlet.java (hunk 2 above)
Comment (isaacrivriv) [reply dev/io.openliberty.wsoc.internal_fat/test-applications/basic.war/src/basic/war/servlet/AddEndpointServlet.java]: Sure no prob I'll make the change

**Comment by isaacrivriv** [comment]
#build

File: dev/io.openliberty.wsoc.internal_fat/fat/src/io/openliberty/wsoc/tests/BasicTest.java (hunk 1 above)
Comment (isaacrivriv) [reply dev/io.openliberty.wsoc.internal_fat/fat/src/io/openliberty/wsoc/tests/BasicTest.java]: Changed so that it matches the other tests we've worked on

**Review by pnicolucci — APPROVED** [review]
Looks good, thanks for making the requested changes!

**Comment by isaacrivriv** [comment]
#libby

**Pull request merged**